File: recovered-source/jadx/sources/io/legado/app/model/analyzeRule/AnalyzeByRegex.java
package io.legado.app.model.analyzeRule;

import java.util.ArrayList;
import java.util.List;
import java.util.regex.Matcher;
import java.util.regex.Pattern;
import kotlin.Metadata;
import kotlin.jvm.internal.Intrinsics;
import me.ag2s.epublib.epub.PackageDocumentBase;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/* JADX INFO: compiled from: AnalyzeByRegex.kt */
/* JADX INFO: loaded from: reader-pro-classes-3.2.14.jar:io/legado/app/model/analyzeRule/AnalyzeByRegex.class */
@Metadata(mv = {1, 5, 1}, k = 1, xi = 48, d1 = {"\u0000&\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0002\b\u0002\n\u0002\u0010 \n\u0002\u0010\u000e\n\u0002\b\u0002\n\u0002\u0010\u0011\n\u0000\n\u0002\u0010\b\n\u0002\b\u0003\bÆ\u0002\u0018\u00002\u00020\u0001B\u0007\b\u0002¢\u0006\u0002\u0010\u0002J3\u0010\u0003\u001a\n\u0012\u0004\u0012\u00020\u0005\u0018\u00010\u00042\u0006\u0010\u0006\u001a\u00020\u00052\f\u0010\u0007\u001a\b\u0012\u0004\u0012\u00020\u00050\b2\b\b\u0002\u0010\t\u001a\u00020\n¢\u0006\u0002\u0010\u000bJ7\u0010\f\u001a\u000e\u0012\n\u0012\b\u0012\u0004\u0012\u00020\u00050\u00040\u00042\u0006\u0010\u0006\u001a\u00020\u00052\f\u0010\u0007\u001a\b\u0012\u0004\u0012\u00020\u00050\b2\b\b\u0002\u0010\t\u001a\u00020\n¢\u0006\u0002\u0010\u000b¨\u0006\r"}, d2 = {"Lio/legado/app/model/analyzeRule/AnalyzeByRegex;", PackageDocumentBase.PREFIX_OPF, "()V", "getElement", PackageDocumentBase.PREFIX_OPF, PackageDocumentBase.PREFIX_OPF, "res", "regs", PackageDocumentBase.PREFIX_OPF, "index", PackageDocumentBase.PREFIX_OPF, "(Ljava/lang/String;[Ljava/lang/String;I)Ljava/util/List;", "getElements", "reader-pro"})
public final class AnalyzeByRegex {

    @NotNull
    public static final AnalyzeByRegex INSTANCE = new AnalyzeByRegex();

    private AnalyzeByRegex() {
    }

    public static /* synthetic */ List getElement$default(AnalyzeByRegex analyzeByRegex, String str, String[] strArr, int i, int i2, Object obj) {
        if ((i2 & 4) != 0) {
            i = 0;
        }
        return analyzeByRegex.getElement(str, strArr, i);
    }

    @Nullable
    public final List<String> getElement(@NotNull String res, @NotNull String[] regs, int index) {
        int groupIndex;
        Intrinsics.checkNotNullParameter(res, "res");
        Intrinsics.checkNotNullParameter(regs, "regs");
        Matcher resM = Pattern.compile(regs[index]).matcher(res);
        if (!resM.find()) {
            return null;
        }
        if (index + 1 == regs.length) {
            ArrayList info = new ArrayList();
            int i = 0;
            int iGroupCount = resM.groupCount();
            if (0 <= iGroupCount) {
                do {
                    groupIndex = i;
                    i++;
                    String strGroup = resM.group(groupIndex);
                    Intrinsics.checkNotNull(strGroup);
                    info.add(strGroup);
                } while (groupIndex != iGroupCount);
            }
            return info;
        }
        StringBuilder result = new StringBuilder();
        do {
            result.append(resM.group());
        } while (resM.find());
        String string = result.toString();
        Intrinsics.checkNotNullExpressionValue(string, "result.toString()");
        int vIndex = index + 1;
        return getElement(string, regs, vIndex);
    }

    public static /* synthetic */ List getElements$default(AnalyzeByRegex analyzeByRegex, String str, String[] strArr, int i, int i2, Object obj) {
        if ((i2 & 4) != 0) {
            i = 0;
        }
        return analyzeByRegex.getElements(str, strArr, i);
    }

    @NotNull
    public final List<List<String>> getElements(@NotNull String res, @NotNull String[] regs, int index) {
        int groupIndex;
        Intrinsics.checkNotNullParameter(res, "res");
        Intrinsics.checkNotNullParameter(regs, "regs");
        Matcher resM = Pattern.compile(regs[index]).matcher(res);
        if (!resM.find()) {
            return new ArrayList();
        }
        if (index + 1 == regs.length) {
            ArrayList books = new ArrayList();
            do {
                ArrayList info = new ArrayList();
                int i = 0;
                int iGroupCount = resM.groupCount();
                if (0 <= iGroupCount) {
                    do {
                        groupIndex = i;
                        i++;
                        String strGroup = resM.group(groupIndex);
                        info.add(strGroup == null ? PackageDocumentBase.PREFIX_OPF : strGroup);
                    } while (groupIndex != iGroupCount);
                }
                books.add(info);
            } while (resM.find());
            return books;
        }
        StringBuilder result = new StringBuilder();
        do {
            result.append(resM.group());
        } while (resM.find());
        String string = result.toString();
        Intrinsics.checkNotNullExpressionValue(string, "result.toString()");
        int vIndex = index + 1;
        return getElements(string, regs, vIndex);
    }
}

File: full-vineflower/io/legado/app/model/analyzeRule/AnalyzeByRegex.kt
package io.legado.app.model.analyzeRule

import java.util.ArrayList
import java.util.regex.Matcher
import java.util.regex.Pattern

public object AnalyzeByRegex {
   public fun getElement(res: String, regs: Array<String>, index: Int = 0): List<String>? {
      val resM: Matcher = Pattern.compile(regs[index]).matcher(res);
      if (!resM.find()) {
         return null;
      } else {
         val var10000: java.util.List;
         if (index + 1 == regs.length) {
            val result: ArrayList = new ArrayList();
            var var11: Int = 0;
            val var8: Int = resM.groupCount();
            val groupIndex: Int;
            if (0 <= var8) {
               do {
                  groupIndex = var11++;
                  val var10001: java.lang.String = resM.group(groupIndex);
                  result.add(var10001);
               } while (groupIndex != var8);
            }

            var10000 = result;
         } else {
            val var10: StringBuilder = new StringBuilder();

            do {
               var10.append(resM.group());
            } while (resM.find());

            val var12: java.lang.String = var10.toString();
            var10000 = this.getElement(var12, regs, index + 1);
         }

         return var10000;
      }
   }

   public fun getElements(res: String, regs: Array<String>, index: Int = 0): List<List<String>> {
      val resM: Matcher = Pattern.compile(regs[index]).matcher(res);
      if (!resM.find()) {
         return new ArrayList<>();
      } else if (index + 1 == regs.length) {
         val var12: ArrayList = new ArrayList();

         do {
            val var14: ArrayList = new ArrayList();
            var var15: Int = 0;
            val var9: Int = resM.groupCount();
            val groupIndex: Int;
            if (0 <= var9) {
               do {
                  groupIndex = var15++;
                  val var11: java.lang.String = resM.group(groupIndex);
                  var14.add(if (var11 == null) "" else var11);
               } while (groupIndex != var9);
            }

            var12.add(var14);
         } while (resM.find());

         return var12;
      } else {
         val result: StringBuilder = new StringBuilder();

         do {
            result.append(resM.group());
         } while (resM.find());

         val info: java.lang.String = result.toString();
         return this.getElements(info, regs, index + 1);
      }
   }
}
